src/main/java/com/revature/validation/JsonValidation.java

```diff
@@ -258,5 +272,65 @@ public void validateQuestionPoolExists(Integer qpId) {
                     "Question with id " + qpId + " does not exist",null);
         }
     }
+    
+    public void validatePersonFields(Person person){
+    	
+    	List<Integer> roleIds = new ArrayList<>();
+		List<PersonRole> roleList = personRoleDao.findAll();
+		System.out.println(roleList);
+		
+		// iterate thru the list of possible personRoles
+		// will be used with for loop below
+		Iterator<PersonRole> iterator = roleList.iterator();
+		while(iterator.hasNext()) {
+			PersonRole role = iterator.next();
+			roleIds.add(role.getId());
+		}
+		
+		if (person.getFirstName() == null) {
+			throw new ConstraintViolationException("Missing required field firstName (String)", null);
+		}
+		
+		if (person.getLastName() == null) {
+			throw new ConstraintViolationException("Missing required field lastName (String)", null);
+		}
+		if (person.getPersonRole() == null) {
+			throw new ConstraintViolationException("Missing required field PersonRole (PersonRole)", null);
+		}
+		
+		// make sure the inputted role is valid role
+		// by matching the personrole id with list values - must match one
+		boolean isValid = false;
+		for (Integer roll : roleIds) {
+				if (person.getPersonRole().getId() == roll) {
+				isValid = true;
+			} 	
+		}	
+		if (!isValid) {
+			throw new ConstraintViolationException("Invalid field personRole (PersonRole)", null);
+		}
+    	
+    }
+    
+    public void validatePersonRole(PersonRole personRole){
+    	
+    	if(personRole == null){
+			throw new ConstraintViolationException("Invalid PersonRole Field", null);
+		}
+    }
+    
+    public Person validatePersonById(int id){
+    	Person p = null;
+		
+		try {
+			
+			p = personDao.findOne(id);
+			
+		} catch (EntityNotFoundException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1166)